### PR TITLE
drop versioneer

### PR DESCRIPTION
**Pull request by ocefpaf**
`versioneer` is deprecated and causes some trouble with downstream packaging. I tried to keep the same versioning scheme with `setuptools_scm` instead.

### CODE REVIEW

File: tests/test_rechunk.py [hunk 1]
```diff
@@ -62,12 +62,9 @@ def test_rechunk_dataset(
 ):
     if target_store.startswith("mapper"):
         target_store = fsspec.get_mapper(str(tmp_path) + target_store)
-    else:
-        target_store = str(tmp_path / target_store)
-
-    if temp_store.startswith("mapper"):
         temp_store = fsspec.get_mapper(str(tmp_path) + temp_store)
     else:
+        target_store = str(tmp_path / target_store)
```
Comment (ocefpaf) [inline tests/test_rechunk.py:67]: This commit is a left-over from #64

Comment (rabernat) [reply tests/test_rechunk.py:67]: I'm not sure I understand your comment. Is it a merge issue?

Comment (ocefpaf) [reply tests/test_rechunk.py:67]: Nope. I was too slow to send it in that PR. Sorry :grimacing:

If you want I can remove it from this PR and do a new one.

Comment (rabernat) [reply tests/test_rechunk.py:67]: No it's fine here.

**Comment by rabernat** [comment]
I think I merged this prematurely. 😭 

There is a mypy test failure: https://github.com/pangeo-data/rechunker/runs/1476473853?check_suite_focus=true#step:8:7

**Pull request merged**